### PR TITLE
Fix runtime cast issue

### LINKED ISSUE #18: Not runtime safe with Dart 2
Run the tests with `--preview-dart-2` (using Dart, like, `> 2.0.0-dev.46` or something), and see failures:

```shell
$ dart --preview-dart-2 test/list_test.dart
...
00:01 +48 -12: sync lists a subdirectory that sometimes exists [E]
  type 'SubListIterable<String>' is not a subtype of type 'List<String>' where
    SubListIterable is from dart:_internal
    String is from dart:core
    List is from dart:core
    String is from dart:core

  package:glob/src/ast.dart 169:33                              SequenceNode.split
  package:glob/src/list_tree.dart 75:31                         new ListTree
  package:glob/glob.dart 150:44                                 Glob.listSync
  test/list_test.dart 340:12                                    syncAndAsync.<fn>.<fn>
  test/list_test.dart 287:24                                    main.<fn>.<fn>
```

### PR DESCRIPTION
**Pull request by srawlins**
Fixes #18.

### CODE REVIEW

**Comment by nex3** [comment]
Please bump the patch version in the pubspec and add a CHANGELOG entry.

File: pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: glob
-version: 1.1.5
+version: 1.1.5+1
```
Comment (nex3) [inline pubspec.yaml]: 1.1.6. We only use build numbers for pre-1.0.0 packages, because the semantics of each version number are shifted down one place.

Comment (srawlins) [reply pubspec.yaml]: Done.

**Comment by srawlins** [comment]
Ugh, sorry this didn't squash, I just added 10% to the commit history 😞

**Pull request merged**